### ISSUE TITLE
strace causing certain program to exit on WSL

### ISSUE THREAD

**Issue by wangqr** [Microsoft/WSL#3695]
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)
Microsoft Windows [Version 10.0.17763.134]

* What you're doing and what's happening:  (Copy&paste specific commands and their output, or include screen shots)
I am `strace`ing some command and the command behaves differently under strace. This happens only under WSL.

* What's wrong / what should be happening instead:
The command should behave the same with and without strace.

* How to reproduce:
Save the following script as `test.py`
```python
#! /usr/bin/env python3

import time
from multiprocessing.pool import ThreadPool
import subprocess

def dummy():
    subprocess.run(['true'])
    time.sleep(100000)


with ThreadPool(processes=1) as tp:
    r = tp.apply_async(dummy, ())
    tp.close()
    r.get()
```
Directly run `./test.py` (on either WSL or linux machine) should output nothing.
When running under strace using `strace -f -o /dev/null ./test.py` on a linux machine, the command behaves the same as without strace. On WSL, the the script exits immediately with
```
sem_wait: Connection timed out
Traceback (most recent call last):
  File "./test.py", line 15, in <module>
    r.get()
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 640, in get
    raise TimeoutError
multiprocessing.context.TimeoutError
```
Here is the strace output on WSL: https://gist.github.com/wangqr/a76d369b572a413e31b0d3f5dbd0fa82

**Comment by wangqr** [comment]
I am actually debugging some other issue that happens only on WSL, described at actionless/pikaur#317. But strace cannot be used due to this issue.

**Comment by wangqr** [comment]
Running under `gdb` behaves the same as under `strace`

**Comment by wangqr** [comment]
Sorry, The previous strace log was generated on Archlinux. Archlinux or Python 3.7 is not needed to reproduce this issue.

This issue can be reproduced with the fresh installed Ubuntu 18.04 from Microsoft Store. The gist is updated to the Ubuntu generated version.

**Comment by therealkenc** [comment]
I haven't looked at this, but spit-balling, this could be related to #3618 which wasn't addressed until 18277. It is always worth giving Insiders a try and hope to get lucky. Doubly so if you happen to know whether this or a similar python pattern was working some time in the past with WSL, or just just popped up recently (for some value of recent). That said there is a curios [comment](https://github.com/actionless/pikaur/issues/317#issuecomment-443074711) in the linked `pikaur` issue: "it saying what each acquiring of the lock was taking more than 2 seconds". I think I have be seeing something similar in an unrelated (unmentionable) scenario that persists on 18290, but I haven't looked into it yet. Thanks for the post.

**Comment by wangqr** [comment]
It seems that this issue has been fixed on 10.0.18290.1000

**Comment by wangqr** [comment]
The pikaur's high CPU usage / slow lock issue still persists on 18290. I think that might be a different issue.

**Comment by therealkenc** [comment]
> It seems that this issue has been fixed on 10.0.18290.1000

Alright. Appreciate the follow-up info. Let's call this dupe #3618 for want. As for the slow lock, if you have a tight repro and are feeling highly motivated, you could open another issue. [If not I wager it will come up soon enough regardless.]
